### ISSUE TITLE
Using hierarchical softmax loss for unbalanced classification

### ISSUE THREAD

**Issue by fishiwhj** [facebookresearch/fastText#440]
According to the fastText [faq.](https://github.com/facebookresearch/fastText/blob/master/docs/faqs.md), 

> Note also that this loss is thought for classes that are unbalanced, that is some classes are more frequent than others.

Can somebody explain this further? Why hierarchical softmax loss is preferred on unbalanced data over other loss functions?

I doubt this since in my problem, a unbalanced text classification task, the hierarchical loss does not work as well as other losses. I have to make the data more balanced to make the algorithm work.

**Comment by MrRace** [comment]
@fishiwhj  Therefore you get the answer by yourself？
